pkg/setup/setup.go

```diff
@@ -289,6 +289,13 @@ func runBackgroundTask(ctx context.Context, config BackgroundTaskConfig, cancel
 	defer cancel()
 	if err := config.Run(ctx, reporter); err != nil {
 		logger.FromContext(ctx).Error("background.error", zap.Error(err), zap.String("job", config.Name))
+	} else {
+		select {
+		case <-ctx.Done():
+			// task finished with no error and context was cancelled, normal shutdown
```

Comment:
We do not need to be so careful with adding log lines when it's about rare cases like shutdown (or startup).
I would suggest to turn this comment into a log (maybe INFO level)